### DISCUSSION TITLE
Automatic subtitle sync tools

### DISCUSSION THREAD

**Discussion by burrinhoMAC** [SubtitleEdit/subtitleedit#8222] [General]
Has anyone tried these automatic subtitle synchronization tools?

FFsubsync
https://github.com/smacke/ffsubsync

SubSync
https://subsync.online/

autosubsync
https://github.com/oseiskar/autosubsync

Are these tools efficient?

**Comment by LeonCheung** [comment]
Do you mean resyncing a file with timecodes already, or automatically generating timecodes for a given transcript?

**Reply by burrinhoMAC** [reply]
Resyncing a file with timecodes already.

**Comment by Purfview** [comment]
Those are pretty dated implementations by the today's standards, I wouldn't expect great efficiency, especially on the audios containing not only speech.

Maybe I could adapt some algo to the today's AI standards.

**Reply by burrinhoMAC** [reply]
I understood.
Thanks.

**Comment by gabrielmagno** [comment]
Having the feature of automatically syncing a subtitle using as reference either the audio of the video or another synced subtitle would be a great feature.

[`ffsubsync`](https://github.com/smacke/ffsubsync) (mentioned by @burrinhoMAC) is probably the most popular tool as of today. It takes into account constant offsets and even handles variation of framerate.

But let me suggest taking a look at [`alass`](https://github.com/kaegi/alass), similar as ffsubsync, but besides constant offsets and variation of framerate it alsos take into account "splits" (due to advertisement). So, for example, taking a subtitle that was created for a video that has the intro/outro of ads could work to sync a subtitle for another release of the video without the ads intro/outro. The algorithm was the author's bachelors thesis.

Personally I've been getting good results with alass. One thing that I noticed is that it works better when the subtitle does not have hearing impaired marks, so for synchronization matters it is good to remove the HI marks (using SubtitleEdit) before the synchronization.
